### PR TITLE
feat: Add team member deletion with validation and confirmation

### PR DESCRIPTION
**Pull request by gentamura**
## Summary
Add team member deletion functionality with proper validation and confirmation UI. The implementation includes safeguards for admin roles and self-deletion scenarios, ensuring team stability.

## Related Issue
N/A

## Changes
- Add server-side team member deletion with validations
  - Prevent removal of last admin
  - Add validation for self-deletion with team count check
  - Implement type-safe role validation
- Add client-side deletion UI components
  - Add confirmation dialog using AlertDialog
  - Add delete button with proper styling
  - Handle loading and error states
- Add proper error handling and feedback
  - Show user-friendly error messages
  - Handle edge cases with appropriate validation

## Testing
- Verified admin role validation
  - Cannot remove last admin
  - Only admin can remove members
- Tested self-deletion scenarios
  - Prevented leaving when it's the only team
  - Confirmed successful deletion with multiple teams
- Tested confirmation dialog
  - Proper display of member information
  - Cancel and confirm actions work as expected

## Other Information
- Uses shadcn/ui AlertDialog for confirmation
- Implements optimistic updates for better UX
- Follows existing team management patterns

### CODE REVIEW

**Comment by gentamura** [comment]
@satococoa review, please :pray:

**Review by satococoa — APPROVED** [review]
Thank you!! 🙆‍♂️ 👍

**Review by shige — APPROVED** [review]
LGTM! 👍

**Comment by gentamura** [comment]
@toyamarinyon review, please 🙏

File: package.json [hunk 1]
```diff
@@ -35,6 +35,7 @@
 		"@opentelemetry/semantic-conventions": "1.28.0",
 		"@paralleldrive/cuid2": "2.2.2",
 		"@radix-ui/react-accordion": "^1.2.0",
+		"@radix-ui/react-alert-dialog": "^1.1.2",
```
Comment (toyamarinyon) [inline package.json]:
```suggestion
		"@radix-ui/react-alert-dialog": "1.1.2",
```

**Review by toyamarinyon — APPROVED** [review]

**Comment by gentamura** [comment]
Thanks for reviewers 🚀

**Pull request merged**